python/cuml/manifold/umap.pyx

```diff
@@ -417,29 +447,37 @@ class UMAP(Base):
                                                      if convert_dtype
                                                      else None))
             fit(handle_[0],
-                < float*> x_raw,
-                < float*> y_raw,
-                < int > self.X_m.shape[0],
-                < int > self.X_m.shape[1],
-                < UMAPParams*>umap_params,
-                < float*>embed_raw)
+                <float*> x_raw,
+                <float*> y_raw,
+                <int> self.n_rows,
+                <int> self.n_dims,
+                <UMAPParams*>umap_params,
+                <float*>embed_raw)
 
         else:
 
             fit(handle_[0],
-                < float*> x_raw,
-                < int > self.X_m.shape[0],
-                < int > self.X_m.shape[1],
-                < UMAPParams*>umap_params,
-                < float*>embed_raw)
+                <float*> x_raw,
+                <int> self.n_rows,
+                <int> self.n_dims,
+                <UMAPParams*>umap_params,
+                <float*>embed_raw)
 
         self.handle.sync()
 
         return self
 
-    def fit_transform(self, X, y=None, convert_dtype=False):
+    def fit_transform(self, X, y=None, convert_dtype=True):
         """Fit X into an embedded space and return that transformed
         output.
+
+        There is a subtle difference between calling fit_transform(X)
+        and calling fit().transform(). Calling fit_transform(X) will
+        train the embeddings on X and return the embeddings. Calling
+        fit(X).transform(X) will train the embeddings on X and then
+        run a second optimization
+        return the embedding after it is trained while calling
```

Comment:
@cjnolet the last few lines here look like they got jumbled. Can you please clarify what they're supposed to say?

Reply:
Yeah, I think that last line was erroneously added in somehow. Thank you for catching this!